cloud_storage_handler/clients/minio_client.py

```diff
@@ -0,0 +1,97 @@
+"""This module provides functionality for managing MinIO client."""
+
+import logging
+
+from foca import Foca
+from minio import Minio
+
+from cloud_storage_handler.api.elixircloud.csh.constants import CshConstants
+from cloud_storage_handler.api.elixircloud.csh.models import MinioConfig
+from cloud_storage_handler.exceptions import ConfigNotFoundError
+from cloud_storage_handler.utils import get_config_path
+
+logger = logging.getLogger(__name__)
+
+
+class MinioConfigurationData:
```

Comment:
If I'm not mistaken, you get all of the functionalities of this class from FOCA for free, once you have defined a custom config (which you did): If the MinIO config is in the app's config file under the `custom` section, it will be parsed and validated against `MinioConfig`. If it's missing (or individual params are missing), it/they will be set according to the defaults defined in the `MinioConfig` model. If you want to access the MinIO config in app context (i.e., when handling a request), which is quite possibly all you ever want, simply access it from the FOCA app instance's config (via `current_app`).

If you need it outside app context (and a FOCA app instance isn't available), just create an app context manually and access it in the same way.

So I don't think this class is needed and only adds maintenance burden and technical debt.